File: 2024/09/part1.py
import io

def solve(line):
    N = len(line)
    arr = []
    k = 0
    for i in range(N):
        for j in range(int(line[i])):
            if i % 2 == 0:
                arr.append(str(k))
            else:
                arr.append(".")
        if i % 2 == 0:
            k += 1

    M = len(arr)
    l = 0
    r = M - 1
    while (l<r):
        while arr[l] != "." and l < r:
            l += 1
        while arr[r] == "." and r > l:
            r -= 1
        arr[l], arr[r] = arr[r], arr[l]
        l += 1
        r -= 1
    
    total = 0
    for i in range(M):
        if arr[i] == ".":
            break
        total += int(arr[i])*i
    return total

with open("input.txt") as file:
    line = str.rstrip(file.readline())
    print(solve(line))

File: 2025/01/part1.rs
use std::{
    fs::File,
    io::{self, BufRead, BufReader},
    path::Path,
};

fn lines_from_file(filename: impl AsRef<Path>) -> io::Result<Vec<String>> {
    BufReader::new(File::open(filename)?).lines().collect()
}

fn main() {
    let lines = lines_from_file("./input.txt").expect("AOC");
    let mut x = 50;
    let mut cnt = 0;
    let m = 100;
    for line in lines {
        let op = line.chars().nth(0);
        let number:i32 = (&line.chars().as_str()[1..]).parse().unwrap();
        if op == Some('R') {
            x += number;
        }
        else {
            x -= number;
        }
        x %= m;
        if x == 0 {
            cnt += 1;
        }
    }
    println!("{cnt}");
}
